vespa/src/main/java/ai/marqo/search/HybridSearcher.java

```
@@ -163,6 +180,36 @@ public Result search(Query query, Execution execution) {
         Integer offset = query.properties().getInteger("offset", 0);
         Integer timeout = query.properties().getInteger("timeout", 1000);
 
+        // Pagination exclusion via DocumentAccess
+        String paginationSchema =
+                query.properties().getString("marqo__hybrid.pagination_schema", null);
+        String paginationHash = query.properties().getString("marqo__hybrid.pagination_hash", null);
+        Integer paginationLimitCutoff =
+                query.properties().getInteger("marqo__hybrid.pagination_limit_cutoff", 600);
+
+        Set<String> idsToExclude = new HashSet<>();
+        AsyncSession docAccess = null;
+        com.yahoo.documentapi.Result paginationDocResult = null;
+        DocumentId docId = null;
+        Set<Integer> paginationStateOffsets = null;
+        Boolean noCache = null;
+        // Get the pagination document if paginationHash and paginationSchema are set and
+        // retrievalMethod is disjunction
+        if (shouldUsePagination(paginationHash, paginationSchema, retrievalMethod)) {
+            try {
+                docId = getPaginationDocumentId(paginationSchema, paginationHash);
+                AsyncParameters asyncParameters = new AsyncParameters();
+                docAccess = documentAccess.createAsyncSession(asyncParameters);
+                // Do not get document is offset is 0, as we do not need to exclude any IDs in this
+                // case.
+                if (offset != 0) {
+                    paginationDocResult = docAccess.get(docId);
+                    logIfVerbose("Search for pagination document: " + docId, verbose);
+                }
+            } catch (Exception e) {
```

Avoid a general exception catch

---

uhh I agree, but at the same time I'm unaware of what exception could be raised here. The purpose of it is more so that pagination is guaranteed to not interfere with search if there is any issue with it.

---

That feels weird. If we don't expect an error here maybe we should just let it raise and return a 500?

---

not sure, but we could. My opinion is that since search can successfully complete with it failing - it shouldn't raise 500

---

Agreed catching Exception seems like a bad idea. You could catch a generic exception class to be safe but not exception which can be anything including bug in your code.